pkg/kubelet/server/stats/types.go

```diff
@@ -70,12 +71,18 @@ type PodStats struct {
 	// Stats pertaining to volume usage of filesystem resources.
 	// VolumeStats.UsedBytes is the number of bytes used by the Volume
 	VolumeStats []VolumeStats `json:"volume,omitempty" patchStrategy:"merge" patchMergeKey:"name"`
+	// User defined metrics that are exposed by containers in the pod. Typically, we expect only one
+	// container in the pod to be exposing user defined metrics. In the event of multiple containers
+	// exposing metrics, they will be combined here.
+	UserDefinedMetrics []UserDefinedMetric `json:"userDefinedMetrics,omitmepty" patchStrategy:"merge" patchMergeKey:"name"`
 }
 
 // ContainerStats holds container-level unprocessed sample stats.
 type ContainerStats struct {
 	// Reference to the measured container.
 	Name string `json:"name"`
+	// The time at which data collection for this container was (re)started.
+	StartTime unversioned.Time `json:"startTime"`
```

Comment:
@timstclair: Is `unversioned.Time` a portable time definition or should we stick to nanoseconds since epoch or something similar?